### DISCUSSION TITLE
build project on CentOS failed

### DISCUSSION THREAD

**Discussion by jijimao** [strongswan/strongswan#2133] [Q&A] [answered]
**System (please complete the following information):**
 - OS: [centos 7.9]
 - Kernel version (if applicable): [3.10.0-1160.el7.x86_64]
 - strongSwan version(s): [5.9.3]
 - Tested/confirmed with the latest version: [yes]

**Describe the bug**

**To Reproduce**
1. ./autogen.sh
2. [root@localhost strongswan-gmalg-merge-5.9.3dr2]# ./configure --prefix=/opt/workspace/Ipsec/strongswan-gmalg-merge/release --enable-kernel-libipsec --enable-gmalg --with-gmalg_interior=yes
configure: WARNING: unrecognized options: --enable-gmalg, --with-gmalg_interior
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '0' is supported by ustar format... yes
checking whether GID '0' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether make supports nested variables... (cached) yes
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for a sed that does not truncate output... /usr/bin/sed
./configure: line 4819: syntax error near unexpected token `dbusdatadir,'
./configure: line 4819: `  PKG_CHECK_VAR(dbusdatadir, dbus-1, datadir, , dbusdatadir="${datarootdir}")'

I check that the dbus-1 lib is installed

why?

**Comment by tobiasbrunner** [comment, marked as answer]
When building from the repository (or also just to rebuild the configure script) you need the tools listed in [HACKING](https://github.com/strongswan/strongswan/blob/master/HACKING) (they have to be installed before running `autogen.sh`). One of them is pkg-config (might be called pkgconfig on CentOS), which provides the `PKG_CHECK_VAR` macro.

**Comment by jijimao** [comment]
thanks，your answer is perfect, I'v compiled and used it!
